virttest/iscsi.py

```diff
@@ -330,14 +330,18 @@ def get_device_name(self):
         Get device name from the target name.
         """
         cmd = "iscsiadm -m session -P 3"
-        pattern = r"%s.*?disk\s(\w+)\s+\S+\srunning" % self.target
+        pattern = r"Lun:\s(\d+).*?scsi\sdisk\s(\w+)\s+State:\srunning"
         device_name = []
         if self.logged_in():
             output = process.run(cmd).stdout_text
             targets = output.split('Target: ')[1:]
             for target in targets:
                 if self.target in target:
-                    device_name = re.findall(pattern, target, re.S)
+                    self.devices = dict(re.findall(pattern, target, re.S))
+                    device_name.append(self.devices["0"])
+                    for lun, device in self.devices.items():
+                        dname = "/dev/%s" % device
+                        self.devices[lun] = dname
```

Comment:
hi Aihua, I just noticed that we need to call get_device_name first, then we can get self.devices, i.e. self.devices depends on get_device_name function, that might not be a good idea. So what about introducing a new interface, like get_devices, which returns the {lun: dev}. In this way, we don't need self.devices, every time we use it, we call the function to get it. For the devices could change(think about that we add a new lun in server, we have to refresh devices)

Reply:
updated, @zhencliu, please help to review again, thanks.